apps/studio/src/pages/sites/[siteId]/settings.tsx

```diff
@@ -1,8 +1,175 @@
+import { useEffect, useState } from "react"
+import { useParams } from "next/navigation"
+import { useRouter } from "next/router"
+import {
+  Button,
+  Center,
+  HStack,
+  Text,
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react"
+import {
+  Infobox,
+  Input,
+  Toggle,
+  useToast,
+} from "@opengovsg/design-system-react"
+
+import { UnsavedSettingModal } from "~/features/editing-experience/components/UnsavedSettingModal"
 import { type NextPageWithLayout } from "~/lib/types"
 import { AdminCmsSidebarLayout } from "~/templates/layouts/AdminCmsSidebarLayout"
+import { trpc } from "~/utils/trpc"
 
 const SiteSettingsPage: NextPageWithLayout = () => {
-  return <div>Settings page</div>
+  const toast = useToast()
+  const router = useRouter()
+  const params = useParams<{ siteId: string }>()
+
+  const notificationMutation = trpc.site.setNotification.useMutation({
+    onSuccess: () => {
+      setNotificationFieldChanged(false)
+      toast({
+        title: "Saved site settings!",
+        description: "Check your site in 5-10 minutes to view it live.",
+        status: "success",
+      })
+    },
+    onError: (error) => {
+      console.log(error)
+      toast({
+        title: "Error saving site settings!",
+        description:
+          "If this persists, please report this issue at support@isomer.gov.sg",
+        status: "error",
+      })
+    },
+  })
+
+  const [notificationEnabled, setNotificationEnabled] = useState(true)
+  const [notificationText, setNotificationText] = useState("")
+  const [notificationFieldChanged, setNotificationFieldChanged] =
+    useState(false)
+
+  const remainingCharacters = 100 - notificationText.length
+  const saveEnabled =
+    notificationFieldChanged &&
+    (!notificationEnabled || notificationText.length !== 0)
+
+  const onClickUpdate = () => {
+    notificationMutation.mutate({
+      siteId: Number(params.siteId),
+      notification: notificationEnabled ? notificationText : "",
+    })
+  }
+
+  const [previousNotification] = trpc.site.getNotification.useSuspenseQuery({
+    siteId: Number(params.siteId),
+  })
+
+  useEffect(() => {
+    setNotificationEnabled(previousNotification !== "")
+    setNotificationText(previousNotification)
+  }, [previousNotification])
```

Comment:
[don't sync state, derive it](https://www.notion.so/opengov/Don-t-sync-state-derive-it-173d074b374644e6bab52e578ad626a3?pvs=3&qid=)

we should not use useEffect wrongly. should we derive state/initialize state instead:

```ts
// notification text always starts with previousNotification
// initialise state with previousNotification
const [notificatonState, useNotificationState] = useState(previousNotification) // <- initialise state
const isNotificationEnabled = !notificationText // <- derive state


```